### PR TITLE
actions: Allow not printing error when an action fails

### PR DESCRIPTION
**Pull request by vikassingh093**
When we are starting the server with superuser action packages show ERROR log message which is not correct

### CODE REVIEW

File: plinth/actions.py [hunk 1]
```diff
@@ -186,8 +186,9 @@ def _run(action, options=None, input=None, async=False, run_as_root=False,
         output, error = proc.communicate(input=input)
         output, error = output.decode(), error.decode()
         if proc.returncode != 0:
-            LOGGER.error('Error executing command - %s, %s, %s', cmd, output,
-                         error)
+            if expecting_error:
+                LOGGER.error('Error executing command - %s, %s, %s', cmd, output,
+                             error)
```
Comment (JoKeyser) [inline plinth/actions.py]: Even if the error was expected, it would still create a (misleading) error message...

**Review by JoKeyser — COMMENTED** [review]
Hi @vikassingh093 , thank you for submitting a PR for issue #1131 .

Your approach is more complex than what I had in mind, and actually won't prevent the misleading error messages (see inline comment).

I think a better approach would be to directly edit the action `subcommand_is_package_manager_busy` in file `actions/packages` to not throw an exception. Instead, it should print to the console 'yes' or 'no' and exit with system code 0 (success), because both are valid answers.
That output string should then be parsed by the function in `plinth/package.py`, `is_package_manager_busy` to return True or False.
Hence no error is logged, and it also gives admins on the console with an easy-to-interpret answer if they execute that action.

Your idea to flag an error as expected or not would also work, but I feel this is over-complicated in this case, and would sort-of hide the real issue.

Hope this is clearer, if not please let me know. Thanks again for taking a stab at this!

**Comment by SunilMohanAdapa** [comment]
@JoKeyser , I had encouraged @vikassingh093 to follow this approach. My first thought was to do it like you suggested.  However, doing that it felt like we are changing a better implementation to something not ideal just because we have an error message that we like to avoid printing. Returning an error code and catching it felt like it is like throwing an exception and catching it, which is many advantages. Returning True and False involves (non-so-ideal) string parsing. Many command line applications take this route of communicating status. Further, we may want to suppress error logging in other cases too and we don't want to change the action file every time.

In the current approach, the method parameter, which I suggested is indeed confusing. We can call it `log_error` (default True). If the code isn't working now (I haven't checked properly), we can fix it. Does this sound okay? Otherwise, we can go with what you suggested.

**Comment by JoKeyser** [comment]
Ha, what a funny little issue this is. My perspective was a more general one on semantics, in the sense that the error/exception channel is reserved for failed execution within the script itself, instead of a valid answer.

I guess I didn't see the tradition of shell scripting where the interpretation of a valid outcome may be common. In case the logging suppression is needed elsewhere it's certainly useful. And I'm also bothered by the inefficient string comparisons. So I can definitely live also with this approach.

I do like the renaming to `log_error`, I guess I expected opposite behavior from the naming in my in-line comment; haven't really checked it yet, could do that soon.

**Comment by SunilMohanAdapa** [comment]
Great!

I think the test case is failing because the file /var/lib/dpkg/lock is not being locked properly. We need to fix that.

**Comment by jvalleroy** [comment]
Repository moved to https://salsa.debian.org/freedombox-team/freedombox.

**Pull request closed without merging**